closer_test.go

```diff
@@ -566,3 +580,41 @@ func TestEndlessGrowth(t *testing.T) {
 		r.NoError(t, children[i].Close())
 	}
 }
+
+func TestCloser_RunCloserRoutine(t *testing.T) {
+	t.Parallel()
+
+	c := closer.New()
+	started := make(chan struct{})
+
+	c.RunCloserRoutine(func() error {
+		close(started)
+		return errors.New("error")
+	})
+
+	select {
+	case <-time.After(3 * time.Second):
+		t.Fatal("timed out")
+	case <-started:
+	}
+
+	c.Close_()
+	r.Error(t, c.CloserError())
+	r.Equal(t, c.CloserError().Error(), "error")
```

Comment:
You could use both `r.EqualError` or even better `r.ErrorIs` instead.

Reply:
Good catch!